### PR TITLE
Chsp physics update, global latlon grid, post changes and CCN-IN followup bugfix

### PR DESCRIPTION
**Pull request by junwang-noaa**
In the commit:
1) CHSP from GFSv16 package is added to gfsphysics and ccpp/physics
2) add capability to output global_latlon to netcdf file
3) land surface model information is added to the history file, post lib is updated to 8.0.6
4) Add CCN-IN followup bugfix in GFS_typedefs.F90 
5)  Bugfix for Intel 18 on wcoss_cray, remove Intel 16 options for gaea
6) CMEPS integration changes

Related PRs:
[ccpp/physics #440](https://github.com/NCAR/ccpp-physics/pull/440) 
[#104](https://github.com/NOAA-EMC/fv3atm/pull/104)
[NEMS #55](https://github.com/NOAA-EMC/NEMS/pull/55)
[ufs-weather-model #108](https://github.com/ufs-community/ufs-weather-model/pull/108)

### CODE REVIEW

File: gfsphysics/GFS_layer/GFS_physics_driver.F90 [hunk 1]
```diff
@@ -4275,7 +4309,7 @@ subroutine GFS_physics_driver                                 &
                           Statein%pgr, Statein%phil, clw, Stateout%gq0,     &
                           Stateout%gt0, Stateout%gu0, Stateout%gv0, rain1,  &
                           kbot, ktop, kcnv, islmsk, Statein%vvl, ncld,      &
-                          Diag%hpbl, hflx, evap, ud_mf, dt_mf, cnvw, cnvc,  &
+                          Diag%hpbl, hflxq, evapq, ud_mf, dt_mf, cnvw, cnvc,&
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_physics_driver.F90:4314]: I am sure you have a reason for it, but I don't understand why this change is only for shalcnv, but not for samfshalcnv (or any other convection scheme)?

File: gfsphysics/physics/ugwp_driver_v0.f [hunk 2]
```diff
@@ -8,6 +8,7 @@ module sso_coorde
        use machine,      only: kind_phys
        real(kind=kind_phys),parameter :: pgwd  = 1._kind_phys
        real(kind=kind_phys),parameter :: pgwd4 = 1._kind_phys
+       logical debugprint = .false.
```
Comment (climbfuji) [inline gfsphysics/physics/ugwp_driver_v0.f]: See comment here: https://github.com/NCAR/ccpp-physics/pull/440#discussion_r415067027

**Review by climbfuji — CHANGES_REQUESTED** [review]

**Comment by junwang-noaa** [comment]
Final RT passed, code is ready for commit after review. Thanks

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by JongilHan66 — APPROVED** [review]
Changes for the canopy heat storage parameterization (CHSP), thermal roughness length, and increased background diffusivity and mixing length in the inversion layers look fine.

**Review by climbfuji — APPROVED** [review]
I did raise one question that remains unanswered (https://github.com/NOAA-EMC/fv3atm/pull/104/files?file-filters%5B%5D=.F90&file-filters%5B%5D=.f&file-filters%5B%5D=.f90&file-filters%5B%5D=.meta&file-filters%5B%5D=dotfile#r414570311), but this shouldn't delay the commit. Otherwise this is a bunch of important changes!

**Comment by junwang-noaa** [comment]
Jongil's answer:  It is because only shalcnv among convection schemes need
kinematic sensible heat flux (hflx) and latent heat flux (evap) for the
cloud base mass flux. The other convection schemes do not use hflx & evap.

File: gfsphysics/GFS_layer/GFS_physics_driver.F90 (hunk 1 above)
Comment (JongilHan66) [reply gfsphysics/GFS_layer/GFS_physics_driver.F90:4314]: It is because only shalcnv among convection schemes needs kinematic sensible heat flux (hflx) and latent heat flux (evap) for the cloud base mass flux. The other convection schemes do not use hflx & evap.

Comment (climbfuji) [reply gfsphysics/GFS_layer/GFS_physics_driver.F90:4314]: Thanks, got it! We need to make sure - I'll create an issue to check - that none of the additional convection schemes in CCPP that are not in IPD use hflx & evap.

Comment (climbfuji) [reply gfsphysics/GFS_layer/GFS_physics_driver.F90:4314]: See https://github.com/NCAR/ccpp-physics/issues/443

**Review by jswhit2 — APPROVED** [review]

**Review by DeniseWorthen — APPROVED** [review]
Looks great. All my changes are present as expected.
